.github/workflows/main.yml

```diff
@@ -0,0 +1,559 @@
+#
+# Copyright (c) 2014-present, The osquery authors
+#
+# This source code is licensed as defined by the LICENSE file found in the
+# root directory of this source tree.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+#
+
+# Due to a limitation in how GitHub Actions works, we can't reference
+# jobs in another file inside the `needs` statement.
+#
+# This configuration file takes care of the Windows, macOS and Linux
+# builds on the x86 platform.
+name: Main
+
+on:
+  push:
+    branches:
+      - '*'
+
+    tags:
+      - 'v*'
+
+  pull_request:
+    branches:
+      - '*'
+
+# If the initial code sanity checks are passing, then one job
+# per [`platform` * `build_type`] will start, building osquery
+# and generating packages that are later attached to the commit
+# (or PR) as build artifacts.
+#
+# The checkout@v2 action removes access to the `.git` folder, so
+# we have to use v1 instead
+jobs:
+
+  # This job performs basic source code check, looking for formatting
+  # issues and missing copyright headers
+  check_source_code:
+    runs-on: ubuntu-18.04
+
+    container:
+      image: trailofbits/osquery:ubuntu-18.04-toolchain-v9
+      options: --privileged --init -v /var/run/docker.sock:/var/run/docker.sock
+
+    steps:
+    # The checkout@v2 action removes access to the `.git` folder, so
+    # we can't use it
+    - uses: actions/checkout@v1
+
+    # This script makes sure that the copyright headers have been correctly
+    # placed on all the source code files
+    - name: Check the copyright headers
+      run: |
+        ./tools/ci/scripts/check_copyright_headers.py
+
+    - name: Setup the build paths
+      shell: bash
+      id: build_paths
+      run: |
+        mkdir -p workspace/build
+        ln -sf "$(pwd)" workspace/src
+
+        echo ::set-output name=SOURCE::$(realpath workspace/src)
+        echo ::set-output name=BINARY::$(realpath workspace/build)
+
+    - name: Configure the project
+      working-directory: ${{ steps.build_paths.outputs.BINARY }}
+      run: |
+        cmake -G "Unix Makefiles" \
+          -DOSQUERY_TOOLCHAIN_SYSROOT:PATH="/usr/local/osquery-toolchain" \
+          -DOSQUERY_ENABLE_FORMAT_ONLY=ON \
+          "${{ steps.build_paths.outputs.SOURCE }}"
+
+    # Formatting is tested against the clang-format binary we ship
+    # with the osquery-toolchain, so this job is only performed once on
+    # a Linux machine.
+    - name: Check code formatting
+      working-directory: ${{ steps.build_paths.outputs.BINARY }}
+      run:
+        cmake --build . --target format_check
```

Comment:
This is after cmake?

Reply:
If understand your question correctly, CMake configuration has to be called for the `format_check` target to be available.

Reply:
👍